var/spack/repos/builtin/packages/kokkos/package.py

```diff
@@ -35,6 +35,7 @@ class Kokkos(CMakePackage, CudaPackage, ROCmPackage):
         'pthread': [False, 'Whether to build Pthread backend'],
         'serial': [True,  'Whether to build serial backend'],
         'rocm': [False, 'Whether to build HIP backend'],
+        'spir': [False, 'Whether to build OneAPI OpenMP SPIR device backend'],
```

Comment:
Note this is not enough and also not the right thing to do for Kokkos. We need to add openmptarget device, and IntelGEN architecture.

Reply:
Yes, this is just a strawman. I have no stake in how this implemented so long as the correct backend is generated and the spack package knows how to communicate that information to cmake. 

Is the intended use of the 'sycl' variant in the package for oneapi, a different backend, or generally? It's never referenced in the package.

Reply:
sycl is a different backend. To target Intel GPUs with OneAPI you can either use their OpenMP Device compiler or the DPC++/SYCL programming model. The latter does not need that spir option, the OpenMPTarget backend does, its in our cmake but not fully exposed in SPACK. Is there an exisitng spack architecture option for Intel GPUs?

Reply:
@becker33 please correct me, but AFAIK spack doesn't track device architecture. 

The SPIR code in my understanding is device independent and JIT-ed, so device arch shouldn't matter for the openmp case, but would thus matter for the SYCL backend? @TApplencourt does this sound right?

Reply:
The device architecture isn't required either `SYCL` either. Without extra flags, you get `JIT` compilation but specifying `KOKKOS_ARCH_INTEL_GEN` enables `AOT` compilation for `Intel Gen9` GPUs.
Since we need extra flags for `OpenMPTarget` to produce `SPIR` code (as opposed to targeting other backends like `CUDA`) and we are mostly interested in `SPIR` code for `Intel GPUs`, we decided to also use `KOKKOS_ARCH_INTEL_GEN` to add the correct flags.